tidy3d/plugins/microwave/array_factor.py

```diff
@@ -28,6 +30,12 @@
 class AbstractAntennaArrayCalculator(Tidy3dBaseModel, ABC):
     """Abstract base for phased array calculators."""
 
+    taper: Union[RectangularTaper, RadialTaper] = pd.Field(
+        None,
+        title="Antenna Array Taper",
+        description="Amplitude weighting of array elements to control main lobe width and suppress side lobes.",
```

Comment:
for fields that allow different types we add `discriminator=TYPE_TAG_STR,` to prevent ambiguous errors for reasons explained above